src/cpp/rtps/builtin/discovery/participant/PDP.cpp

```diff
@@ -1698,6 +1700,123 @@ void PDP::add_builtin_security_attributes(
 
 #endif // HAVE_SECURITY
 
+void PDP::local_participant_attributes_update_nts(
+        const RTPSParticipantAttributes& new_atts)
+{
+    // Update user data
+    auto participant_data = getLocalParticipantProxyData();
+    participant_data->m_userData.data_vec(new_atts.userData);
+
+    bool announce_locators = !mp_RTPSParticipant->is_intraprocess_only();
+    if (!announce_locators)
+    {
+        // If we are intraprocess only, we do not need to update locators
+        return;
+    }
+
+    // Clear all locators
+    participant_data->metatraffic_locators.unicast.clear();
+    participant_data->metatraffic_locators.multicast.clear();
+    participant_data->default_locators.unicast.clear();
+    participant_data->default_locators.multicast.clear();
+
+    // Update default locators
+    for (const Locator_t& loc : new_atts.defaultUnicastLocatorList)
+    {
+        participant_data->default_locators.add_unicast_locator(loc);
+    }
+    for (const Locator_t& loc : new_atts.defaultMulticastLocatorList)
+    {
+        participant_data->default_locators.add_multicast_locator(loc);
+    }
+
+    // Update metatraffic locators
+    for (const auto& locator : new_atts.builtin.metatrafficUnicastLocatorList)
+    {
+        participant_data->metatraffic_locators.add_unicast_locator(locator);
+    }
+    if (!new_atts.builtin.avoid_builtin_multicast || participant_data->metatraffic_locators.unicast.empty())
+    {
+        for (const auto& locator : new_atts.builtin.metatrafficMulticastLocatorList)
+        {
+            participant_data->metatraffic_locators.add_multicast_locator(locator);
+        }
+    }
+
+    fastdds::rtps::network::external_locators::add_external_locators(*participant_data,
+            new_atts.builtin.metatraffic_external_unicast_locators,
+            new_atts.default_external_unicast_locators);
+}
+
+void PDP::update_endpoint_locators_if_default_nts(
+        const std::vector<BaseWriter*>& writers,
+        const std::vector<BaseReader*>& readers,
+        const RTPSParticipantAttributes& old_atts,
+        const RTPSParticipantAttributes& new_atts)
+{
+    // Check if default locators have changed
+    const auto& old_default_unicast = old_atts.defaultUnicastLocatorList;
+    const auto& old_default_multicast = old_atts.defaultMulticastLocatorList;
+    const auto& new_default_unicast = new_atts.defaultUnicastLocatorList;
+    const auto& new_default_multicast = new_atts.defaultMulticastLocatorList;
+
+    // Early return if there is no change in default unicast locators
+    if ((old_default_unicast == new_default_unicast) &&
+            (old_default_multicast == new_default_multicast))
+    {
+        return;
+    }
```

Comment:
Why remaining locators are not considered?

Reply:
Default locators are the ones that are copied to DATA(w) and DATA(r), so they are the only ones relevant for this method.